code/model/ErrorPage.php

```diff
@@ -274,47 +301,65 @@ public function fieldLabels($includerelations = true) {
 		
 		return $labels;
 	}
-	
+
 	/**
-	 * Returns an absolute filesystem path to a static error file
-	 * which is generated through {@link publish()}.
-	 * 
-	 * @param int $statusCode A HTTP Statuscode, mostly 404 or 500
-	 * @param String $locale A locale, e.g. 'de_DE' (Optional)
+	 * Get error content for the given status code. Only returns pre-cached content
+	 * stored either in the asset store or in the static path (if enabled).
 	 *
+	 * @param int $statusCode A HTTP Statuscode, typically 404 or 500
 	 * @return string
 	 */
-	public static function get_filepath_for_errorcode($statusCode, $locale = null) {
-		if (singleton('ErrorPage')->hasMethod('alternateFilepathForErrorcode')) {
-			return singleton('ErrorPage')-> alternateFilepathForErrorcode($statusCode, $locale);
```

Comment:
It's removed, because it's not necessary for the core to respect specific implementations.

My intention was to provide `updateErrorFilename` as sam suggested, as an alternative to this. All that was done here was to remove the locale parameter, but that can be retrieved by `Translatable::get_current_locale()` in the extension (rather than in the ErrorPage.php itself) in any case.